### DISCUSSION TITLE
Problem with useToast.

### DISCUSSION THREAD

**Discussion by svrem** [zeit-ui/react#536] [Q&A] [answered]
I was trying to use useToast but I wasn't able to make it work. I imported useToast from @geist/react and then just copied the code from the doc site. But when I clicked the button it didn't work. This is my code:

```
import { useToasts, Button } from "@geist-ui/react";

const toast = () => {
  const [toasts, setToast] = useToasts();
  const click = () =>
    setToast({ text: "The Evil Rabbit jumped over the fence." });
  return <Button onClick={click}>Show Toast</Button>;
};

export default toast;
```

I greatly appreciate your help...

Btw:
I don't really know if I'm good here asking this question. If so please let me know and I will delete this post.

**Comment by ghost** [comment, marked as answer]
You need to use `useToasts` in a react component. `toast` should be replaced by `Toast`.

```jsx
import { useToasts, Button } from '@geist-ui/react';

const Toast = () => {
  const [, setToast] = useToasts();
  const onClick = () => setToast({ text: 'The Evil Rabbit jumped over the fence.' });

  return <Button onClick={onClick}>Show Toast</Button>;
};

export default Toast;
```

**Reply by svrem** [reply]
Hey, 
Thanks for your response. I think I did something wrong. I copied your code but it still wasn't working. Did you want for me to make a new file and then paste the code in and then import it from another file. Or should it work if I just paste the code in a file.

**Reply by ghost** [reply]
> Did you want for me to make a new file and then paste the code in and then import it from another file?

This really depends on your use case. But yes, that code should work fine even if you create a new file (eg. `toast.js`) and import it in another file.

> Or should it work if I just paste the code in a file?

The main objective was to show `useToasts` hook can only be used in _react components_. In your code the component name was `toast`, but keeping in mind the react component rules, your component name should start with a **capital letter** (ie. `Toast`).

```diff
import { useToasts, Button } from "@geist-ui/react";

- const toast = () => {
+ const Toast = () => {
  const [toasts, setToast] = useToasts();
  const click = () =>
    setToast({ text: "The Evil Rabbit jumped over the fence." });
  return <Button onClick={click}>Show Toast</Button>;
};

- export default toast;
+ export default toast;
```

If you have any doubts about using it in a specific _react component_, just paste your code in comments.

**Reply by svrem** [reply]
Even with these updates to the code (eg. toast => Toast). I still have the problem that when I click the button the toast doesn't show up. I tested if the click function worked with a console log and it did. So I think the setToast doesn't work. Do you know how to fix this?

**Reply by ghost** [reply]
Can you send your code below or your project in a shareable link?

**Reply by svrem** [reply]
``` 
import { useToasts, Button } from "@geist-ui/react";

const Toast = () => {
  const [toasts, setToast] = useToasts();

  const click = () => {
    setToast({ text: "The Evil Rabbit jumped over the fence." });
  };

  return <Button onClick={click}>Show Toast</Button>;
};

export default Toast;
```

**Reply by ghost** [reply]
I used this exact code on my local machine and it worked. Here is a codesandbox [link](https://codesandbox.io/s/vigilant-keller-yfcsr?file=/src/App.js).

**Reply by svrem** [reply]
Oh, I think that I know what I'm doing wrong. I don't use React I use Next.js. All the other things worked on there so I thought Toasts would work too. But apparently I doesn't.

I'm sorry for wasting your time. But thank you for all the help.

**Reply by ghost** [reply]
I actually used next.js on my local machine to check. All these components do work in next.js. In fact the geist website is made with next.js. You might be doing something wrong. Did you import `<GeistProvider>` and `<CssBaseline>` in your `_app.jsx`?

**Reply by svrem** [reply]
Oh I don't. Give me a second to check.

**Reply by svrem** [reply]
````
import "../styles/globals.css";
import { GeistProvider, CssBaseline } from "@geist-ui/react";

function MyApp({ Component, pageProps }) {
  return (
    <GeistProvider>
      <CssBaseline />
      <Component {...pageProps} />
    </GeistProvider>
  );
}

export default MyApp;
````

This is my _app.js. It still isn't working. Do I import it correctly?

**Reply by ghost** [reply]
Your `_app.js` looks fine. Can you share your project link?

**Reply by svrem** [reply]
Well, I'm not doing it in CodeSandbox and I'm not very familiar with it. How can I upload my project to it?

**Reply by ghost** [reply]
No need for CodeSandbox. Just commit your project on github and share the repository link.

**Reply by svrem** [reply]
Oh great will do.

**Reply by svrem** [reply]
I hope you can see this project. Else let me please know.

https://github.com/svrem/My-project

**Reply by ghost** [reply]
I can't see it. It must be private.

**Reply by svrem** [reply]
Can you try again?

**Reply by ghost** [reply]
I see the project now. Can you upload the `package.json` too?

**Reply by svrem** [reply]
oh of course

**Reply by svrem** [reply]
If you reload it's there

**Reply by svrem** [reply]
Oh

**Reply by svrem** [reply]
I have found the problem. Apperently you can't name your file toast.js. When I tested it on codesandbox it worked and I toughed that was weird. Then I tried to rename my file. And that worked.

Thank you very much for all the help.

**Reply by ghost** [reply]
It works. Here are the steps I followed.

```shell
git clone https://github.com/svrem/My-project.git
cd .\My-project\Sharing\
yarn install
yarn dev
```

Open your browser to [http://localhost:3000](http://localhost:3000) but it would appear as a 404 as you don't have an `index.js` file. Now go to [http://localhost:3000/toast](http://localhost:3000/toast) to see the button, and when clicked, a toast appears in the bottom right corner.

![image](https://user-images.githubusercontent.com/66206863/117700945-49939280-b1e4-11eb-9f27-cea28cfe35b5.png)

**Reply by svrem** [reply]
Weird, when I do those steps it works for me also. But it doesn't work on my main project. But what did work is just to rename the file. So it is fixed. Thank you for all the help!
